### PR TITLE
A missing requirement and a backported import for TimeoutError

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

File: clkhash/backports.py [hunk 1]
```diff
@@ -108,6 +108,7 @@ def _p2_unicode_reader(unicode_csv_data, dialect=csv.excel, **kwargs):
         TimeoutError = globals()['__builtins__']['TimeoutError']
     except:
         pass    # pypy3 raises TypeError: 'module' object is not subscriptable
+        TimeoutError = TimeoutError
```
Comment (nbgl) [inline clkhash/backports.py]: Why is this necessary?

Comment (hardbyte) [reply clkhash/backports.py]: Because we need `TimeoutError` to be attached to the `clkhash.backports` module scope. :-/

Yes supporting all these Python versions is annoying.

Comment (nbgl) [reply clkhash/backports.py]: That makes sense. As far as I can tell though `TimeoutError = globals()['__builtins__']['TimeoutError']` will always throw, so the entire `try`/`except` block is redundant with the exception of `TimeoutError = TimeoutError`.

I think a neater way of rewriting lines 106-113 would be ```python
try:
    TimeoutError = TimeoutError
except NameError:
    # We're in Python < 3.3.
    TimeoutError = OSError
```

Comment (hardbyte) [reply clkhash/backports.py]: Oh wow, so we merged something that always throws? Opps!

I just assumed that on Python3.6+ that the module must have been subscriptable and pypy3 was just being a bit pedantic or annoying.

Comment (hardbyte) [reply clkhash/backports.py]: Oh and you can't just do `TimeoutError = TimeoutError` because that'd be too easy. (try it)

But this should work (I think) `globals()['__builtins__'].TimeoutError`

File: tests/test_backports.py [hunk 2]
```diff
@@ -7,10 +7,18 @@
 import io
 import unittest
 
+import clkhash.backports
 from clkhash.backports import (int_from_bytes, re_compile_full,
                                unicode_reader, strftime)
```
Comment (nbgl) [inline tests/test_backports.py]: Mixing import levels. Either use `import foo` or `from foo import bar`.

File: clkhash/backports.py [hunk 3]
```diff
@@ -105,9 +105,10 @@ def _p2_unicode_reader(unicode_csv_data, dialect=csv.excel, **kwargs):
 
 if sys.version_info > (3, 2):
     try:
-        TimeoutError = globals()['__builtins__']['TimeoutError']
+        TimeoutError = globals()['__builtins__'].TimeoutError
     except:
-        pass    # pypy3 raises TypeError: 'module' object is not subscriptable
+        # pypy3 raises TypeError: 'module' object is not subscriptable
+        TimeoutError = TimeoutError
```
Comment (nbgl) [inline clkhash/backports.py]: You shouldn't need the `try`/`except` at all. If `TimeoutError` is in builtins then it will never throw. If it is not, then your except clause will throw.

File: clkhash/backports.py [hunk 4]
```diff
@@ -105,9 +105,10 @@ def _p2_unicode_reader(unicode_csv_data, dialect=csv.excel, **kwargs):
 
 if sys.version_info > (3, 2):
     try:
-        TimeoutError = globals()['__builtins__']['TimeoutError']
+        TimeoutError = globals()['__builtins__'].TimeoutError
```
Comment (nbgl) [inline clkhash/backports.py]: No need to access `globals`. Use `__builtins__.TimeoutError`. Even better, just `TimeoutError` since it's a builtin.

File: clkhash/backports.py [hunk 5]
```diff
@@ -105,9 +105,10 @@ def _p2_unicode_reader(unicode_csv_data, dialect=csv.excel, **kwargs):
 
 if sys.version_info > (3, 2):
```
Comment (nbgl) [inline clkhash/backports.py]: I suggest `try: TimeoutError = TimeoutError` instead of explicitly checking for versions. It's slightly safer as you don't have to keep track of the changelogs.

File: tests/test_backports.py (hunk 2 above)
Comment (hardbyte) [reply tests/test_backports.py]: I think this is better as I want to test existence of particular attributes of the `backports` module which shouldn't be done at test import but rather at test execution - hence the `import clkhash.backports`. However the other backport tests are arguably cleaner by importing into the top level scope.

File: clkhash/backports.py (hunk 3 above)
Comment (hardbyte) [reply clkhash/backports.py]: Good point. I've revised this to be much simpler - let's hope travis/jenkins agree.

File: clkhash/backports.py (hunk 4 above)
Comment (hardbyte) [reply clkhash/backports.py]: Can't use the last one as it isn't a builtin in all versions but it is a good point that it was unnecessarily complex.

**Comment by hardbyte** [comment]
Thanks for the review Jakub - I'll rebase and request you to have another look.

**Comment by nbgl** [comment]
Looks like Mypy is failing. I think clkhash/backports.py:107 needs to change from
```python
    TimeoutError = __builtins__.TimeoutError
```
to
```python
    TimeoutError = __builtins__.TimeoutError  # type: type
```
